request-manager/grapher/grapher.go

```diff
@@ -692,6 +695,12 @@ func (o *Grapher) newNoopNode(name string, nodeArgs map[string]interface{}) (*No
 
 // newNode creates a node for the given job j
 func (o *Grapher) newNode(j *NodeSpec, nodeArgs map[string]interface{}) (*Node, error) {
+	// Make a copy of the nodeArgs before this node gets created and potentially
+	// adds additional keys to the nodeArgs.
+	originalArgs := map[string]interface{}{}
+	for k, v := range nodeArgs {
+		originalArgs[k] = v
+	}
```

Comment:
If one of the job args is something like a map (so only the reference is copied into originalArgs), and the job modifies it during Create, this will end up modifying the arg in job.Args too. Is this ok and/or intentional?